### PR TITLE
Implements operation delegation feature

### PR DESCRIPTION
**Pull request by mdanish98**
- Addresses one part of #166 

Signed-off-by: Mohammad Ghazanfar Ali Danish <ghazanfar.danish@iese.fraunhofer.de>
Co-authored-by: Marie Witt <marie.witt@iese.fraunhofer.de>

### CODE REVIEW

File: basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/operation/delegation/OperationDelegationSubmodelRepository.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	@Override
+	public void deleteSubmodelElement(String submodelId, String idShortPath) throws ElementDoesNotExistException {
+		decorated.deleteSubmodelElement(submodelId, idShortPath);
+	}
+
+	@Override
+	public OperationVariable[] invokeOperation(String submodelId, String idShortPath, OperationVariable[] input) throws ElementDoesNotExistException {
+		SubmodelElement sme = getSubmodelElement(submodelId, idShortPath);
+		
+		if (!(sme instanceof InvokableOperation))
```
Comment (FrankSchnicke) [inline basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/operation/delegation/OperationDelegationSubmodelRepository.java:135]: If there is no qualifier, just call decorated.invokeOperation(..)

File: basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/Readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+This feature is enabled by default and to disable this the following property should be defined:
+
+```
+basyx.submodelrepository.feature.operation.delegation.enabled = false
+```
+
+To create an operation delegation, a Submodel that has an Operation(-SubmodelElement) must be present:
+-	The Operation has neither input, output nor inout variables
```
Comment (FrankSchnicke) [inline basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/Readme.md:25]: Why shouldn't the operation have input/output variables?

**Review by FrankSchnicke — CHANGES_REQUESTED** [review]

File: basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/operation/delegation/OperationDelegationSubmodelRepository.java (hunk 1 above)
Comment (mdanish98) [reply basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/src/main/java/org/eclipse/digitaltwin/basyx/submodelrepository/feature/operation/delegation/OperationDelegationSubmodelRepository.java]: Refactored this as per your suggestion.

File: basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/Readme.md (hunk 2 above)
Comment (mdanish98) [reply basyx.submodelrepository/basyx.submodelrepository-feature-operation-delegation/Readme.md]: This was unnecessary, hence removed it.

**Review by FrankSchnicke — APPROVED** [review]

**Pull request merged**